packages/dds/task-manager/src/taskManager.ts

```diff
@@ -478,10 +496,14 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
         this.connectionWatcher.on("disconnect", disconnectHandler);
         this.completedWatcher.on("completed", checkIfCompleted);
 
-        if (!this.connected) {
+        if (!this.isAttached()) {
+            // Simulate auto-ack in detached scenario
+            assert(this.runtime.clientId !== undefined, "clientId is undefined");
```

Comment:
This assert probably hits in the clicker example for the reason mentioned above (subscribing on a detached container)?